caproto/threading/client.py

```diff
@@ -950,22 +955,19 @@ def get_pvs(self, *names, priority=0, connection_state_callback=None,
             with self.pv_cache_lock:
                 try:
                     pv = self.pvs[(name, priority)]
-                    new_instance = False
                 except KeyError:
                     pv = PV(name, priority, self, connection_state_callback,
                             access_rights_callback)
                     names_to_search.append(name)
                     self.pvs[(name, priority)] = pv
                     self.pvs_needing_circuits[name].add(pv)
-                    new_instance = True
 
-            if not new_instance:
-                if connection_state_callback is not None:
-                    pv.connection_state_callback.add_callback(
-                        connection_state_callback)
-                if access_rights_callback is not None:
-                    pv.access_rights_callback.add_callback(
-                        access_rights_callback)
+            if connection_state_callback is not None:
```

Comment:
Callbacks should be double-checked - it's possible that this won't function as you're hoping. 

That is, just because you add a new callback to an already-existing PV, doesn't mean that the callback will be executed and the new listener will be able to see the connectivity/access rights status.

Reply:
I think you just identified the critical bug here...

Reply:
See fb090a8